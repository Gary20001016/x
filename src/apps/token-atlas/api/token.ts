import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

// store
import { addMiddleware } from '../../../store';

// types
import { TokenAtlasInfoApiResponse, TokenAtlasGraphApiResponse, TrendingTokens, BlockchainList } from '../../../types/api';

export const tokenInfoApi = createApi({
    reducerPath: 'tokenInfoApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://token-nubpgwxpiq-uc.a.run.app' }),
    endpoints: (builder) => ({
        getTokenInfo: builder.query<TokenAtlasInfoApiResponse, { asset: string; blockchain: string; symbol: string; }>({
            query: ({ asset, blockchain, symbol }) => {
                const blockchainParam = blockchain !== undefined ? `&blockchain=${blockchain}` : '';
                return `?asset=${asset}&symbol=${symbol}${blockchainParam}`;
            },
        }),
    })
});

export const tokenGraphApi = createApi({
    reducerPath: 'tokenGraphApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://tokenpricehistory-nubpgwxpiq-uc.a.run.app' }),
    endpoints: (builder) => ({
        getTokenGraph: builder.query<TokenAtlasGraphApiResponse, { blockchain?: string; asset: string; from: number; to?: number; }>({
            query: ({ blockchain, asset, from, to }) => {
                const toParam = to !== undefined ? `&to=${from * 1000}` : '';
                const blockchainParam = blockchain !== undefined ? `&blockchain=${blockchain}` : '';
                const assetParam = asset.split(' ')[0];
                return `?asset=${assetParam}&from=${from * 1000}${toParam}${blockchainParam}`;
            },
        }),
    })
})

export const trendingTokensApi = createApi({
    reducerPath: 'trendingTokensApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://trendingtokens-nubpgwxpiq-uc.a.run.app' }),
    endpoints: (builder) => ({
        getTrendingTokens: builder.query<TrendingTokens, void>({
            query: () => '',
        }),
    }),
});

export const blockchainsListApi = createApi({
    reducerPath: 'blockchainsListApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://blockchains-nubpgwxpiq-uc.a.run.app' }),
    endpoints: (builder) => ({
        getBlockchainsList: builder.query<BlockchainList, void>({
            query: () => '',
        }),
    }),
});

/**
 * Add this to the store
 */
addMiddleware(tokenInfoApi);
addMiddleware(tokenGraphApi);
addMiddleware(trendingTokensApi);
addMiddleware(blockchainsListApi);

export const { useGetTokenInfoQuery } = tokenInfoApi;
export const { useGetTokenGraphQuery } = tokenGraphApi;
export const { useGetTrendingTokensQuery } = trendingTokensApi;
export const { useGetBlockchainsListQuery } = blockchainsListApi;
