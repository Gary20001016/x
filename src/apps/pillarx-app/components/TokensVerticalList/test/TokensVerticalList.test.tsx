import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

// components
import TokensVerticalList from '../TokensVerticalList';

// types
import { TokenData } from '../../../../../types/api';

jest.mock('../../HorizontalToken/HorizontalToken', () => 'HorizontalToken');
jest.mock('../../Typography/Body', () => 'Body');

describe('<TokensVerticalList />', () => {
  const mockData: TokenData[] = [
    { 
      id: 1,
      name: 'Token A', 
      symbol: 'TKA',
      contracts: [{ address: '0x1', blockchain: 'Ethereum' }],
      logo: 'logoA.png',
      trending_score: 90,
      platforms: [{ name: 'Platform A', rank: 1, weight: 0.5 }],
      price_change_24h: 5,
      pair: 'TKA/USD'
    },
    { 
      id: 2,
      name: 'Token B', 
      symbol: 'TKB',
      contracts: [{ address: '0x2', blockchain: 'Ethereum' }],
      logo: 'logoB.png',
      trending_score: 80,
      platforms: [{ name: 'Platform B', rank: 2, weight: 0.4 }],
      price_change_24h: -3,
      pair: 'TKB/USD'
    },
  ];

  it('renders correctly and matches snapshot for left position', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <TokensVerticalList position="left" data={mockData} />
        </MemoryRouter>
      )
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly and matches snapshot for right position', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <TokensVerticalList position="right" data={mockData} />
        </MemoryRouter>
      )
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct number of HorizontalToken components', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <TokensVerticalList position='left' data={mockData} />
      </MemoryRouter>
    ).toJSON() as ReactTestRendererJSON;

    const tokensList = tree.children as ReactTestRendererJSON[];
    
    expect(tokensList.length).toBe(mockData.length);
  });

  it('passes the correct props to HorizontalToken components', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <TokensVerticalList position='left' data={mockData} />
      </MemoryRouter>
    ).toJSON() as ReactTestRendererJSON;

    const linkElement = tree.children as ReactTestRendererJSON[];
    
    linkElement.forEach((linkElement, index) => {
      const tokenElement = linkElement.children as ReactTestRendererJSON[];

      tokenElement.forEach((token) => {
        expect(token.props.tokenIndex).toBe(index + 1);
        expect(token.props.tokenName).toBe(mockData[index].name);
        expect(token.props.tokenSymbol).toBe(mockData[index].symbol);
        expect(token.props.tokenValue).toBeUndefined();
        expect(token.props.percentage).toBeUndefined();
        expect(token.props.isLast).toBe(index === mockData.length - 1);
      })
    });
  });
});
