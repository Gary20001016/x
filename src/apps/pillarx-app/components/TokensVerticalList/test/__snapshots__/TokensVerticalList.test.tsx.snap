// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensVerticalList /> renders correctly and matches snapshot for left position 1`] = `
<div
  className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pr-10 border-r-[3px] border-[#1F1D23]"
>
  <a
    href="/token-atlas?asset=Token A&symbol=TKA"
    onClick={[Function]}
  >
    <HorizontalToken
      isLast={false}
      tokenIndex={1}
      tokenLogo="logoA.png"
      tokenName="Token A"
      tokenSymbol="TKA"
    />
  </a>
  <a
    href="/token-atlas?asset=Token B&symbol=TKB"
    onClick={[Function]}
  >
    <HorizontalToken
      isLast={true}
      tokenIndex={2}
      tokenLogo="logoB.png"
      tokenName="Token B"
      tokenSymbol="TKB"
    />
  </a>
</div>
`;

exports[`<TokensVerticalList /> renders correctly and matches snapshot for right position 1`] = `
<div
  className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pl-10"
>
  <a
    href="/token-atlas?asset=Token A&symbol=TKA"
    onClick={[Function]}
  >
    <HorizontalToken
      isLast={false}
      tokenIndex={4}
      tokenLogo="logoA.png"
      tokenName="Token A"
      tokenSymbol="TKA"
    />
  </a>
  <a
    href="/token-atlas?asset=Token B&symbol=TKB"
    onClick={[Function]}
  >
    <HorizontalToken
      isLast={true}
      tokenIndex={5}
      tokenLogo="logoB.png"
      tokenName="Token B"
      tokenSymbol="TKB"
    />
  </a>
</div>
`;
